### ISSUE TITLE
version 2.2.1 compile error using vs2013 on win7-64

### ISSUE THREAD

**Issue by cppbitman** [warmcat/libwebsockets#886]
Two kinds of error
(1)libwebsockets-2.2.1\test-server\../lib/libwebsockets.h(36): error C2054: expected '(' to follow 'inline'
libwebsockets-2.2.1\test-server\../lib/libwebsockets.h(36): error C2085: 'lws_is_be' : not in formal parameter list

(2)error LNK2019: unresolved external symbol snprintf referenced in function lws_ssl_get_error_string

when I remove the keyword "inline " of the code snippet as following in header file libwebsockets.h, error (1) disappear, but error (2) still exist.

static inline int lws_is_be(void) {
	const int probe = ~0xff;
	return *(const char *)&probe;
}

**Comment by lws-team** [comment]
Same problem exists in v2.2-stable?

**Comment by cppbitman** [comment]
yes, exactly the same problems

**Comment by lws-team** [comment]
Well, it builds on appveyor and other windows users, but those are 32-bit IIUI.

So 1) is a bit nasty, again if I understood it, 32-bit msvc is OK with

```
static LWS_INLINE int lws_is_be(void) {
```

but your 64-bit toolchain wants to see

```
static int LWS_INLINE lws_is_be(void) {
```

If we want to patch around it, it should be like `#ifdef WIN64`, or something else?

Problem 2) should be solved by a patch just pushed on master and v2.2-stable.

**Comment by lws-team** [comment]
The snprintf error is solved since the last comment also on v2.2-stable.

There are many instances of the pattern with static and LWS_INLINE then return type in libwebsockets.h, eg

```
static LWS_INLINE int
lws_vfs_file_close(lws_fop_fd_t *fop_fd)
```

I dunno why your toolchain picks on `lws_is_be()`, except that it's designed to be optimized out at compile time (gcc does so).  But if it just complains about inline on that function it sounds like a toolchain problem.
